File: Data Structures & Algorithms/max-consecutive-ones/submission-1.py
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        count = 0
        curr = 0
        for num in nums:
            if num != 1:
                curr = 0
            
            if num == 1:
                curr += 1
                count = max(count, curr)

        return count
            

File: Data Structures & Algorithms/number-of-one-bits/submission-1.java
class Solution {
    public int hammingWeight(int n) {
        int count = 0;
        while (n > 0) {
            if ((n & 1) == 1) {
                count++;
            }
            n = n >> 1; // same as n / 2
        }
    return count;
    }
}
